### PR TITLE
Better Dragging Behaviour

### PR DESCRIPTION
**Pull request by maximaximal**
Make viewer only draggable if user clicked into the viewer to start the drag. This way, the viewer doesn't move when a drag is initiated somewhere else on the page (e.g. text selection).

### CODE REVIEW

**Review by JacobRBlomquist — COMMENTED** [review]
Looks good to me.
I'm hoping to return to this project soon to add side views/on-screen controls etc.
I'll let @johnbeech  have the final say on this PR.
Thanks for your work!

**Comment by maximaximal** [comment]
Hi Jacob! Thanks for the positive feedback. I'll be happy to contribute some more in the future, as written in my other comments around the project :)

File: src/components/user-inputs.js [hunk 1]
```diff
@@ -9,8 +9,24 @@ function registerOn (browserWindow, { viewState }) {
    * Called whenever the mouse is pressed
    */
   function mousePressed () {
+    if (browserWindow.mouseX < 0 || browserWindow.mouseX > viewState.canvasWidth) {
+      return
+    } if (browserWindow.mouseY < 0 || browserWindow.mouseY > viewState.canvasHeight) {
+      return
+    }
```
Comment (johnbeech) [inline src/components/user-inputs.js:16]: Suggest a bit of a refactor, thoughts?
```js
  function isMouseOutOfBounds({ mouseX, mouseY }) {
    return (mouseX < 0 || mouseX > viewState.canvasWidth || mouseY < 0 || mouseY > viewState.canvasHeight)
  }

  /**
   * P5 MousePressed
   *
   * Called whenever the mouse is pressed
   */
  function mousePressed () {
    if (isMouseOutOfBounds(browserWindow)) {
      return
    }

    viewState.clickX = browserWindow.mouseX
    viewState.clickY = browserWindow.mouseY
    viewState.dragActive = true
  }
```

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+
+[*.js]
+indent_style = space
+indent_size = 2
+
```
Comment (johnbeech) [inline .editorconfig:12]: Not sure editorconfig needs to be part of this PR, (or needed at all) - using `standard` to enforce linting.

**Review by johnbeech — COMMENTED** [review]
Tested locally via DFMA website loading local build:
https://mkv25.net/dfma/map-7993-severedcoils?viewer=html5&version=localhost

Happy with approach to change - made a suggested refactor to make mousePressed() function more readable.

File: src/components/user-inputs.js (hunk 1 above)
Comment (maximaximal) [reply src/components/user-inputs.js:16]: I like this refactor, yes. Makes the change more planned and readable!

File: .editorconfig (hunk 2 above)
Comment (johnbeech) [reply .editorconfig:12]: If you want to make it a separate PR that's fine, but I'd prefer for it to not be part of this PR - apologies for the bureaucracy.

File: src/components/user-inputs.js (hunk 1 above)
Comment (johnbeech) [reply src/components/user-inputs.js:16]: Are you ok making that change on your branch?

**Review by johnbeech — APPROVED** [review]
Decided I'll merge this and fix changes from master.

**Comment by johnbeech** [comment]
Tested in production here:
- https://mkv25.net/dfma/map-7993-severedcoils?viewer=html5&version=latest

Haven't prepared a tagged release yet.

**Comment by maximaximal** [comment]
Thank you very much! Was going to change my PR today, was tied up with some student union the last two days... I'll reset my fork to the new version and will try to do multiple PRs if multiple things come up again. Sorry for the additional reviewing work.

**Pull request merged**